### PR TITLE
[libmikmod]fix missing dll bug.

### PR DESCRIPTION
**Pull request by JackBoosY**
See #3020.
The Cmakelists.txt is missing the statement to install the compiled dll, and the configuration that should not be installed is installed. So I corrected it.

### CODE REVIEW

**Comment by JackBoosY** [comment]
@grdowns  can you help me to solve Linux problem? Thanks.

File: ports/libmikmod/portfile.cmake [hunk 1]
```diff
@@ -17,7 +17,13 @@ vcpkg_download_distfile(ARCHIVE
     FILENAME "libmikmod-3.3.11.1.tar.gz"
     SHA512 f2439e2b691613847cd0787dd4e050116683ce7b05c215b8afecde5c6add819ea6c18e678e258c0a80786bef463f406072de15127f64368f694287a5e8e1a9de
 )
+
 vcpkg_extract_source_archive(${ARCHIVE})
```
Comment (Rastaban) [inline ports/libmikmod/portfile.cmake]: Can you switch this to use vcpkg_extract_source_archive_ex and use the PATCHES argument?  That will also allow you to remove the `set` on line 14.

File: ports/libmikmod/portfile.cmake [hunk 2]
```diff
@@ -17,7 +17,13 @@ vcpkg_download_distfile(ARCHIVE
...
+vcpkg_apply_patches(
+    SOURCE_PATH ${SOURCE_PATH}
+    PATCHES
+        ${CMAKE_CURRENT_LIST_DIR}/fix-missing-dll.patch
```
Comment (Rastaban) [inline ports/libmikmod/portfile.cmake]: You can just write `fix-missing-dll.patch` because the default directory is `${CMAKE_CURRENT_LIST_DIR}`

**Comment by JackBoosY** [comment]
@Rastaban  all done, thanks for your advice.

**Pull request merged**